policies/lib/tree/30_generic_methods/user_secondary_group.cf

```diff
@@ -82,87 +72,63 @@ bundle agent user_secondary_groups(login, groups, force) {
 
     any::
       "pass3" expression => "pass2";
       "pass2" expression => "pass1";
       "pass1" expression => "any";
 
 
-    pass3::
-      "error_subcall"    expression => "${subcall_prefix[${group_list}]}_error";
-      "repaired_subcall" expression => "${subcall_prefix[${group_list}]}_repaired";
-      "success_subcall"  expression => and( "${subcall_prefix[${group_list}]}_kept",
-                                            not("error_subcall|repaired_subcall")
-                                          );
-
-      # ${c_id}_{error|repaired|success} are already defined if in enforce
-      # There are redefined here for clarity
-      "${c_id}_error"    expression => or( "!user_exists",
-                                           "group_not_found",
-                                           "error_subcall",
-                                           "${c_id}_error"
-                                         );
-
-      "${c_id}_repaired" expression => and( "user_exists",
-                                            "!group_not_found",
-                                            or( "repaired_subcall",
-                                                "${c_id}_repaired"
-                                              )
-                                          );
-
-      "${c_id}_success"  expression => and( "user_exists",
-                                            "!group_not_found",
-                                            or( "success_subcall",
-                                                "${c_id}_success"
-                                              )
-                                          );
-
   users:
     pass2::
       "${login}"
         policy => "present",
         groups_secondary => { @{group_list} },
-        classes => classes_generic_two("${class_prefix}", "${c_id}"),
-        if => and( "user_exists",
-                   "!group_not_found",
-                   "purge_other_groups"
-                 );
+        classes => classes_generic_two("${class_prefix}", "${report_data.method_id}"),
+        if =>  "purge_other_groups.user_and_groups_exist";
 
   methods:
-    # Without enforce, fallback to user_group
-    pass2.!pass3.!purge_other_groups.user_exists.!group_not_found::
-     "${report_data.method_id}"   usebundle => disable_reporting;
-     "${report_data.method_id}"   usebundle => user_group(
-                                                 "${login}",
-                                                 "${group_list}"
-                                               );
-     "${report_data.method_id}"   usebundle => enable_reporting,
-                                         if => "should_report";
+    # Without enforce, fallback to user_group, we need to use another bundle since we are using an iterator (group_list) here
+    pass2.fallback_user_group::
+      "${report_data.method_id}" usebundle => run_user_group("${login}","${group_list}", "${inner_result}");
```

Comment:
This most likely requires a sanitizing before the _classes_copy as inner_result will be used multiple times